src/Metadata/Delete.php

```diff
@@ -91,8 +86,79 @@ public function __construct(
         ?string $name = null,
         $provider = null,
         $processor = null,
-        array $extraProperties = []
+        array $extraProperties = [],
     ) {
-        parent::__construct(self::METHOD_DELETE, ...\func_get_args());
+        parent::__construct(
+            self::METHOD_DELETE,
```

Comment:
I'd prefer that we use named arguments here as positional ones may break and it'll be harder to maintain. I wrote https://3v4l.org/OFhko as an experiment to do it magically, not sure it's better.

Reply:
Cleaned up: https://3v4l.org/KSd7t (by @dunglas )

Reply:
optimization:  https://3v4l.org/qqvuK

Reply:
Hard to read and probably resource intensive (reflection), that looks like unnecessary complexity to me. Better avoid such specificity IMHO

Reply:
I'm in favor of having something automated like that. It will prevent a whole class of errors (forgetting to pass a new parameter to the parent's constructor, and this already happened), and will make the maintenance easier.

The overhead shouldn't be significant because it's the reflection is only used one time per type, and only at build time (at runtime, the cache is used).

👍 on my side.

Reply:
> It will prevent a whole class of errors (forgetting to pass a new parameter to the parent's constructor

Can't those be covered by php-cs-fixer and/or static analyzers as part of the CI?

Reply:
Also a problem we currently have (see `$paginationViaCursor`) and is not solved by this approach is that type mismatches are only found at runtime.

Reply:
This should be caught when building the cache at compile time.

Reply:
@chalasr not a big deal as this is cached and unless you call `newInstance` attributes are not computed.

Also the problem of `$paginationViaCursor` is something else we need to cover. 

Anyways its either this approach or the manual one where we use named arguments.

Reply:
How would you implement it on the `Operation` -> `HttpOperation` -> `Get` hierarchy?

Default values cannot be used because they’re not passed as arguments, so `new Get` would try to pass `null` to `$method` and fail.

Reply:
we can actually handle default values: https://3v4l.org/h5FGN